app/core/api_errors.py

```
@@ -0,0 +1,529 @@
+def _sanitize_error_value(value: Any) -> Any:
+    if isinstance(value, str):
+        prefix = f"{_ERROR_MARKER}\n" if _ERROR_MARKER in value else ""
+        return f"{prefix}{client_safe_error_message(value)}"
+    if isinstance(value, (dict, list, tuple)):
+        return sanitize_client_payload(value, error_context=True)
+    if value is None:
+        return None
+    return client_safe_error_message(value)
+
+
+def _is_error_envelope(payload: Any) -> bool:
```

Ruff (0.16.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `payload`

(ANN401)